### PR TITLE
Better messaging for takeover amounts on publish page

### PR DESCRIPTION
**Pull request by seanyesmunt**
More clarity for takeover amounts on claims

Show amount needed to win a claim when editing. Ex: "You have a claim at xxx, for 5 more LBC you will control it"

### Changes
I created a new component `NameHelpText` for variable help text under the name input. I moved `BidHelpText` under the bid input (which is probably where it should have been in the first place)

#### Before
screen shot 2018-09-24 at 8 24 53 pm

#### After
screen shot 2018-09-24 at 8 23 39 pm

### CODE REVIEW

File: src/renderer/component/publishForm/internal/bid-help-text.jsx [hunk 1]
```diff
@@ -58,16 +59,35 @@ class BidHelpText extends React.PureComponent<Props> {
...
     }
-
-    return winningBidForClaimUri ? (
+    
+    if (!winningBidForClaimUri && !winndingBidForClaimWithChannel) {
+      return __('Any amount will give you the winning bid.');
+    }
+    
+    const generateTakeOverMessage = (amount, uri) => {
+      return `${__('A deposit greater than')} ${amount} ${__('is needed to win')} ${uri}`
```
Comment (tzarebczan) [inline src/renderer/component/publishForm/internal/bid-help-text.jsx]: Is "win" the right word here? Maybe control or mention vanity? We use it in a few places...

File: src/renderer/component/publishForm/internal/bid-help-text.jsx [hunk 2]
```diff
@@ -58,16 +59,35 @@ class BidHelpText extends React.PureComponent<Props> {
...
+    }
+    
+    let content;
+    if (winndingBidForClaimWithChannel) {
+      content = `Winning bid is ${winndingBidForClaimWithChannel}`
+    }
+    
+    // <div>
+    //   {__('However, you can still get')}{' '}
+    //   {(winndingBidForClaimWithChannel && winningBidForClaimUri) ? __('these URLs for any amount')}
```
Comment (tzarebczan) [inline src/renderer/component/publishForm/internal/bid-help-text.jsx]: I'm assuming you'll add this back in :)

File: src/renderer/component/publishForm/internal/bid-help-text.jsx (hunk 1 above)
Comment (seanyesmunt) [reply src/renderer/component/publishForm/internal/bid-help-text.jsx]: I like control. I think we can definitely improve this copy.

File: src/renderer/component/publishForm/internal/bid-help-text.jsx (hunk 2 above)
Comment (seanyesmunt) [reply src/renderer/component/publishForm/internal/bid-help-text.jsx]: Yes. This PR still needs a lot of cleanup.

**Comment by seanyesmunt** [comment]
@tzarebczan I think this is working correctly. Before cleaning up/renaming stuff could you give this a test?

I'm open to adding new messages/re-wording it but I mainly want to see if it's working.

How it should work:
```
uri on publish page:
lbry://@sean/sean

messages:
A deposit greater than 1.234 is needed to control lbry://sean
Any amount will let you control lbry://@sean/sean
```

If there is no channel in the uri:
```
uri on publish  page:
lbry://sean

messages:
A deposit greater than 1.234 is needed to control lbry://sean
```

If there are no matches for the short uri or uri with channel:
```
uri on publish page:
lbry://@somerandomchannel/somerandomclaim

messages:
Any amount will let you control lbry://somerandomclaim
Any amount will let you control lbry://@somerandomchannel/somerandomclaim
```

**Comment by tzarebczan** [comment]
the messaging about both urls in the last case might be confusing, I think it's enough only to tell them `Any amount will let you control lbry://somerandomclaim` - what do you think? 

`Any amount will let you control lbry://@sean/sean`  what if someone takes over @sean? 

I still haven't run through it, will give that a shot today.

**Comment by seanyesmunt** [comment]
I think it would be ok to only show the short url

**Comment by tzarebczan** [comment]
@seanyesmunt we also discussed showing vanity top bids when editing claims as part of this PR.

File: src/renderer/component/publishForm/internal/name-help-text.jsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        'You are currently editing this claim. If you change the URL, you will need to reselect a file.'
+      );
+    } else if (uri && myClaimForUri) {
+      const editUri = buildURI({
+        contentName: myClaimForUri.name,
+        claimId: myClaimForUri.claim_id,
+      });
+
+      nameHelpText = (
+        <React.Fragment>
```
Comment (skhameneh) [inline src/renderer/component/publishForm/internal/name-help-text.jsx]: Can also use the short syntax (empty tags):
```
<>
  {/* markup here */}
</>
```

Comment (seanyesmunt) [reply src/renderer/component/publishForm/internal/name-help-text.jsx]: Do we need to do something with webpack for that to work? Doesn't seem to build properly when I do that.

**Comment by tzarebczan** [comment]
`If you bid more than 110082.30000001 LBC, when somone navigates to lbry://one it will load your published content.`
How can we make it clearer that they can still publish to this URL and have their content at the exact URL? We can't show them the claim id because it's not known until the transaction is created. 

(just noticed the misspelling on `someone`)

**Comment by seanyesmunt** [comment]
@tzarebczan "However, you can get a longer version of this URL for any price"?

**Comment by tzarebczan** [comment]
"However, you can get a longer version of this URL for any bid"?

**Comment by tzarebczan** [comment]
We should also add similar bid help to the channel creation process (this is opened as a separate issue I believe).

**Pull request merged**